test/object-store/sync/sync_test_utils.cpp

```diff
@@ -331,6 +332,36 @@ struct FakeLocalClientReset : public TestClientReset {
 
 #if REALM_ENABLE_AUTH_TESTS
 
+static void wait_for_object_to_persist(std::shared_ptr<SyncUser> user, const AppSession& app_session,
```

Comment:
TestClientReset has to compile without REALM_ENABLE_AUTH_TESTS or REALM_ENABLE_SYNC, so I think I prefer to keep this as a static function in the cpp file to avoid having to deal the conditionally compiled sync types in the header.